### PR TITLE
Update documentation on profiles and usage examples

### PR DESCRIPTION
**Pull request by bentsherman**
Issue #120

This PR adds documentation on config profiles as well as some basic command-line examples for running GEMmaker in different environments. I also added `docker` and `singularity` profiles so that these modes can be enabled via profiles in lieu of the `-with-docker` and `-with-singularity` flags.

### CODE REVIEW

**Comment by JohnHadish** [comment]
I am of the opinion that we should completely scrap the `-with-docker` and `-with-singularity` command line options if we have them available in the `nextflow.config`. I think that having two options would just confuse the user. It would also be better for reproducibility if they were just contained within the `nextflow.config` instead of being command line arguments.

I suppose that we might want to consider getting rid of the command line arguments for `-profiles ____` as well and just adding an option into the config. I don't feel that there is a compelling reason to leave them on the command line.

**Comment by bentsherman** [comment]
@JohnHadish I think you might be misunderstanding the way that these command-line options work. They are inherent to nextflow, not GEMmaker. That is why it's important that we still support the use of `-with-docker` and `-with-singularity` because this method of enabling docker / singularity is __generic to all nextflow workflows__. Same with the `-profiles` option, it is the only way to specify profiles in nextflow and it is common to all workflows.

**Comment by JohnHadish** [comment]
Yep, I understand that `-with-docker` and `-with-singularity` are generic to all workflows, but we do not have to access them via the command line. I personally think that we should remove documentation about the command line arguments and just have them be in the `nextflow.config`. I think it is confusing to have documentation on both. Since the users will more than likely use our `nextflow.config` file as a starting point, it would be easiest to just have the parameter contained in their instead of having another command line argument that they need to add. If we had both, I think that it could result in confusion with what was actually running.

From the nextflow documentation:
```
If you want to avoid entering the Singularity image as a command line parameter, you can define it in the Nextflow configuration file. For example you can add the following lines in the nextflow.config file:

process.container = '/path/to/singularity.img'
singularity.enabled = true
In the above example replace /path/to/singularity.img with any Singularity image of your choice.
```
[source](https://www.nextflow.io/docs/latest/singularity.html)

As for the profiles, it looks like I was incorrect in assuming that we could have this as a parameter as well, it looks like default nextflow behavior forces us to use a command line argument.

File: docs/configuration.rst [hunk 1]
```diff
@@ -159,9 +163,11 @@ The following sections are intended to give the user access to intermediary file
   }
 
 
+
 Output Parameters Descriptions
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-**Default Parameters Typically fine, skip this section unless you need intermediary files**
+
+**Default Parameters Typically fine, skip this section unless you need intermediate files**
```
Comment (spficklin) [inline docs/configuration.rst]: This isn't a proper English sentence. Why is "Default Parameters Typically"  capitalized? It tells the user to skip the section unless they need intermediate files, but will the reader know what that means?  I think if this is just a notice perhaps we should use the RST note construct instead?

**Comment by spficklin** [comment]
In response to the --with-docker or --with-singularity I understand the idea of trying to keep it simple and not confuse the readers. Readers who may be confused or want help will find these options in other online documentation.  So, I actually find it helpful to have both methods described in the documentation.  My preference would be to leave them in the documentation.

File: docs/configuration.rst (hunk 1 above)
Comment (spficklin) [reply docs/configuration.rst]: I think we can just take this line out.  Intermediate files are discussed below this.

**Comment by bentsherman** [comment]
@spficklin This branch is ready to merge. The CI is failing only because the hisat2 download site is down right now, I think it's usually the reason why our CI builds fail randomly.

**Comment by JohnHadish** [comment]
Approved

**Comment by spficklin** [comment]
Thanks  @bentsherman and @JohnHadish .  merging....

**Comment by spficklin** [comment]
oh... it was already merged...

**Pull request merged**